### ISSUE TITLE
Anderson-Darling bad results

### ISSUE THREAD

**Issue by andresfduque** [scipy/scipy#6306]
It seems that Anderson-Darling test, comparing a gumbel distribution is producing wrong results. They seem to be related with the sample size, cause for small sizes it works.
When it is performed compared to a normal distribution it works perfect.

y = np.random.gumbel(0, 1, 10000)
a2 = stats.anderson(y, "gumbel")
a2 = inf

Thank you.

**Comment by tpudlik** [comment]
A little more detail: on SciPy 0.16.1, the error message I get is

    scipy/stats/morestats.py:1214: RuntimeWarning: divide by zero encountered in log
      A2 = -N - sum((2*i - 1.0) / N * (log(z) + log(1 - z[::-1])), axis=0)
    AndersonResult(statistic=inf, critical_values=array([ 0.473,  0.636,  0.755,  0.875,  1.036]), significance_level=array([ 25. ,  10. ,   5. ,   2.5,   1. ]))

**Comment by WarrenWeckesser** [comment]
This can be fixed with the appropriate use of the `logcdf` and `logsf` methods of the distributions used in the `anderson` function.  I'll get started on a pull request.

**Comment by WarrenWeckesser** [comment]
A pull request with a fix is here: https://github.com/scipy/scipy/pull/6309

**Comment by WarrenWeckesser** [comment]
With the fix in #6309:

```
In [5]: from scipy import stats

In [6]: y = np.random.gumbel(0, 1, 10000)

In [7]: a2, crit, sig = stats.anderson(y, 'gumbel')

In [8]: a2
Out[8]: 570.82637965706999
```
